keadm/app/cmd/edge/join.go

```diff
@@ -87,6 +89,8 @@ func addJoinOtherFlags(cmd *cobra.Command, joinOptions *types.JoinOptions) {
 
 	cmd.Flags().StringVar(&joinOptions.KubeEdgeVersion, types.KubeEdgeVersion, joinOptions.KubeEdgeVersion,
 		"Use this key to download and use the required KubeEdge version")
+	joinOptions.EdgeYaml.EdgeD.Version = joinOptions.KubernetesVersion
```

Comment:
This should be joinOptions.KubeEdgeVersion and not joinOptions.KubernetesVersion. Please fix it.